src/api/0x/index.ts

```diff
@@ -0,0 +1,136 @@
+import { OrderKind } from '@cowprotocol/contracts'
+import log from 'loglevel'
+import { SupportedChainId } from '../../constants/chains'
+import { Context } from '../../utils/context'
+import { getTokensFromMarket } from '../../utils/market'
+import { toErc20Address } from '../../utils/tokens'
+
+import BaseApi from '../base'
+
+import { Options, PriceQuoteParams } from '../cow/types'
+import ZeroXError, { logPrefix, ZeroXErrorResponse } from './error'
+import { ERC20BridgeSource, MatchaOptions, MatchaPriceQuote } from './types'
+
+// GPV2Settlement
+// https://etherscan.io/address/0x9008d19f58aabd9ed0d60971565aa8510560ab41
+const AFFILIATE_ADDRESS = '0x9008D19f58AAbD9eD0D60971565AA8510560ab41'
+const EXCLUDED_SOURCES: ERC20BridgeSource[] = []
+const API_VERSION = 'v1'
+
+function get0xUrls(): Partial<Record<SupportedChainId, string>> {
+  // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+  // See https://0x.org/docs/api#introduction
+  return {
+    [SupportedChainId.MAINNET]: 'https://api.0x.org/swap',
+  }
+}
+
+function _getMatchaChainId(chainId: SupportedChainId): SupportedChainId | null {
+  switch (chainId) {
+    // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+    // See https://0x.org/docs/api#introduction
+    // but we only support mainnet of that list so..
+    case SupportedChainId.MAINNET:
+      return chainId
+
+    default:
+      // Unsupported network
+      return null
+  }
+}
+
+function _optionsToSearchUrl(options: MatchaOptions) {
+  const { excludedSources, affiliateAddress } = options
+
+  const excludedSourceString = excludedSources.length > 0 ? excludedSources.join(',') : ''
+  const optionsMap = Object.entries({ excludedSources: excludedSourceString, affiliateAddress }).map(
+    ([option, value]) => `${option}=${value}`
+  )
+  return optionsMap.join('&')
+}
+
+async function _handleQuoteResponse<T = unknown, P extends PriceQuoteParams = PriceQuoteParams>(
+  response: Response,
+  params?: P
+): Promise<T> {
+  if (!response.ok) {
+    const errorObj: ZeroXErrorResponse = await response.clone().json()
```

Comment:
why is the clone needed here?

Reply:
because the body is used twice (here and L62)